openrouteservice/src/main/java/org/heigit/ors/matrix/MultiTreeMetricsExtractor.java

```diff
@@ -95,6 +96,11 @@ else if (graph instanceof QueryGraph) {
 		maxEdgeId = chGraph.getAllEdges().length();
 	}
 
+	public void setSwap(boolean swap){
+		this.swap = swap;
+//		this.reverseOrder = !swap;
+	}
+
```

Comment:
Could you maybe explain why there is need of the two separate flags `reverseOrder` and `swap`? I tried following the logic but got quickly lost 🙈

Reply:
`reverseOrder` is a copy from the original MetricsExtractor. I do not know what it does exactly.
`swap` is explained in the code: if number of sources larger than number of destinations, we replace them with eath other, as this is faster to calculate. This results in backwards calculations and backwards Subgraph building and backwards extraction from the multi tree.

Reply:
I'm still somehow confused about the exact relationship of these two parameters. As originally `reverseOrder` was initialized to `true`, after introducing `swap` I would expect it to be `reverseOrder = !swap` as in the commented out line. Was there any issue with this approach?

Reply:
Ok, maybe I got it. Actually `reverseOrder` seems to be relevant only for shortcuts, and apperently it's value is always set new before it's used. Then the initial value shouldn't matter after all.

Reply:
Would you mind removing the commented out L101 and marking the conversation as resolved? Cheers!

Reply:
tbh my approach to adding swap to this was trial and error, as I am still not 100% certain what `reverseOrder` does. Importantly, swap does exactly what it is supposed to do and works from start of algorithm to extraction of paths.

Reply:
Yeah, I see your point. I was just confused about the seemingly similar parametes, but apparently `reverseOrder` does sth else 🤷‍♂️  Hauptsache es funktioniert! 😅